Generator/Sources/NeedleFramework/Utilities/ProcessUtilities.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Copyright (c) 2018. Uber Technologies
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// Check if the sourcekit deamon process is running by searching through
+/// all processes without ttys.
+var isSourceKitRunning: Bool {
+    // Select processes without controlling ttys in jobs format.
+    let result = ProcessUtilities.execute(process: "/bin/ps", withArguments: ["-xj"])
+    return result.lowercased().contains("sourcekit")
```

Comment:
On my machine, I see 2 that match : 
```
rudro 55866     1 55866      0    0 Ss     ??   43:46.67 /Applications/Xcode.app/Contents/SharedFrameworks/SourceKit.framework/Versions/A/XPCServices/com.apple.dt.SKAgent.xpc/Contents/MacOS/com.apple.dt.SKAgent
rudro 77372     1 77372      0    0 Ss     ??   15:17.26 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/sourcekitd.framework/Versions/A/XPCServices/SourceKitService.xpc/Contents/MacOS/SourceKitService
```

Should we tighten this to `sourcekitservice` ?